### PR TITLE
chore(waku): Enable message missing verification flag

### LINKED ISSUE #20879: Enable new waku message verification flag
# Problem

Once this status-go PR https://github.com/status-im/status-go/pull/5570#discussion_r1690794119 is merged, the recommendation is that we update mobile code to enable the flags `EnableMissingMessageVerification` and `EnableStoreConfirmationForMessagesSent` because that will give a better UX overall.

It would be good to have this for 2.30 if possible, but we are pushing for 2.31 realistically.

### PR DESCRIPTION
**Pull request by ilmotta**
- Fixes https://github.com/status-im/status-mobile/issues/20879

### Summary

As recommended [here](https://github.com/status-im/status-go/pull/5570#discussion_r1692345230) and [here](https://github.com/status-im/status-go/pull/5570#discussion_r1690794119), we set the Waku config flag `WakuV2EnableMissingMessageVerification` to true. The flag can't be toggled in the UI because there's no supporting endpoint in status-go at the moment.

### Steps to test

In order for the flag to be used you have to create a new Profile. If you login with an existing one, it will be the same as if the flag was disabled (as in `develop`).

It would be good to get this PR in 2.30 as it can improve the UX, but only if we deem it safe.

Areas worth testing that may be impacted:

- Overall message reliability should be the same or better.
- Verify data consumption is more or less the same in a mobile network.
- Verify app's behavior in wifi and mobile networks.

status: ready

### CODE REVIEW

**Review by qfrank — APPROVED** [review]
I saw the comments:
https://github.com/status-im/status-go/pull/5570/files#r1690794119
https://github.com/status-im/status-go/pull/5570/files#r1692345230

I'm wondering if we should enable `WakuV2EnableStoreConfirmationForMessagesSent` by default now? because it's disabled by default and also disabled for mobile V1 User @ilmotta

**Comment by ilmotta** [comment]
> I saw the comments: https://github.com/status-im/status-go/pull/5570/files#r1690794119 https://github.com/status-im/status-go/pull/5570/files#r1692345230
> 
> I'm wondering if we should enable `WakuV2EnableStoreConfirmationForMessagesSent` by default now? because it's disabled by default and also disabled for mobile V1 User @ilmotta

@qfrank, I'm not sure what's best, but we have a PR from @cammellos https://github.com/status-im/status-mobile/pull/20715#issue-2401741958 that was merged last month and from the explanation it looks like enabling `WakuV2EnableStoreConfirmationForMessagesSent` can cause some UX problems, so I think it's better we keep it disabled by default.

**Comment by ilmotta** [comment]
@churik I think we better postpone this PR for 2.30.1. cc @richard-ramos

**Comment by churik** [comment]
3 tests are failed because of SauceLabs (they are really unstable lately), 1 because of https://github.com/status-im/status-mobile/issues/21151

**Comment by churik** [comment]
@ilmotta it is not compatible with current desktop - so when mobile device send contact request, the desktop user doesn't recieve it.

CR was sent on 16:45, logs of sender (mobile): [logs (4).zip](https://github.com/user-attachments/files/16965348/logs.4.zip)
CR was not received in 2 mins: [geth_not_r copy.log](https://github.com/user-attachments/files/16965363/geth_not_r.copy.log)

All is working with nightly11/09 (CR arrives immidiately on desktop) 
Desktop build: [master 11/09](https://ci.infra.status.im/job/status-desktop/job/master/3994/)
If these changes should be done on desktop too, we should implement them together, currently I'm blocking the PR.

**Comment by ilmotta** [comment]
> @ilmotta it is not compatible with current desktop - so when mobile device send contact request, the desktop user doesn't recieve it.
> 
> CR was sent on 16:45, logs of sender (mobile): [logs (4).zip](https://github.com/user-attachments/files/16965348/logs.4.zip) CR was not received in 2 mins: [geth_not_r copy.log](https://github.com/user-attachments/files/16965363/geth_not_r.copy.log)
> 
> All is working with nightly11/09 (CR arrives immidiately on desktop) Desktop build: [master 11/09](https://ci.infra.status.im/job/status-desktop/job/master/3994/) If these changes should be done on desktop too, we should implement them together, currently I'm blocking the PR.

@churik, I tried now with the public Desktop 2.30 beta and with this PR's build and the CR sent from mobile arrived immediately on desktop and the CR acceptance confirmation arrived 1min later on mobile. Then I tested with a new profile and the CR sending & confirmation arrived instantly. Other chat messages also are working fine. With these test cases, I think this is enough for me to think the PR is working as expected _when waku is friendly_ and that the PR is compatible with desktop without any further changes on their end.

When appropriate, could you re-check this PR?

**Comment by churik** [comment]
Posting the results here:

Chat | Test data | Network | Start | End | Cellular data(start) | Cellular data(end) | Expected | Actual
-- | -- | -- | -- | -- | -- | -- | -- | --
1-1 | 10 messages | 5G | 16:20 | 16:22 | 899 | 900 | all delivered |  ✅ 
community | 10 re-transmitted messages | offline -> 5G | 16:22 | 16:27 | 900 | 908 | all delivered |   ✅  delivered, but should re-login to make the app to see the status change
community | 20 messages | 5g -> wi-fi | 16:27 | 16:27 | 908 | 912 | all delivered | ✅  IOS does not detect the chage in network and continues to use cellular for some time, [reference](https://github.com/status-im/status-mobile/pull/20730#issuecomment-2233423479) 
community | 10 messages | offline -> 5G | 16:27 | 16:29 | 912 | 914 | should not be fetched | ✅  should re-login to make the app to see the status change (so to receive new messages) [here](https://github.com/status-im/status-mobile/pull/20730#issuecomment-2230798805) and [here](https://github.com/status-im/status-mobile/issues/21172)
1-1 | 20 rentransmitted messages | offline -> 5G | 16:29 | 16:33 | 914 | 919 | all arrived | 🔴 only 13/20 arrived in 35 mins
---- | background | 5G | 16:33 | 17:01 | 919 | 929 |   |  


The bottom line:
- I don't see any improvements / neither regression taking into account https://github.com/status-im/status-mobile/issues/21172 and https://github.com/status-im/status-mobile/issues/21173
- the data is roughly the same https://github.com/status-im/status-mobile/pull/20730#issuecomment-2239048521 (no way to check it precisely from what I can get)
- re-transmitted messages sometimes are not delivered (consider it as a waku issue?); waited for 35 mins.


I leave desicion on to merge it or not to you, as frankly I don't see a lot of improvements and I'm not sure what from the above requires more investigation in particular case.

**Comment by ilmotta** [comment]
Most excellent analysis @churik 👏🏼 I don't want to merge if there's no measurable benefit, but maybe there's a theoretical benefit that's hard to reproduce. @richard-ramos, based on the results above https://github.com/status-im/status-mobile/pull/21005#issuecomment-2356237316, what's your recommendation in terms of enabling the flag `WakuV2EnableMissingMessageVerification`?

**Comment by richard-ramos** [comment]
IMO should be merged. Desktop has been using it for a while and has not reported issues with it either, and the benefits while not immediatly tangible, should still lead to less missing messages as it will constantly attempt to missing messages from storenodes.

**Comment by ilmotta** [comment]
> IMO should be merged. Desktop has been using it for a while and has not reported issues with it either, and the benefits while not immediatly tangible, should still lead to less missing messages as it will constantly attempt to missing messages from storenodes.

Thank you @richard-ramos, I didn't know Desktop was using this flag for a while.

With this added evidence, I'll rebase and merge @churik
